packages/manager/apps/key-management-system/src/api/GET/apiv2/services.ts

```diff
@@ -93,3 +96,49 @@ export const getListingIcebergV2 = async ({
     return null;
   }
 };
+
+export type OKMSOptions = {
+  pagination: PaginationState;
+  sorting: ColumnSort;
+};
+
+/**
+ *  Get okms listing with iceberg V2
+ */
+
+export const getListingIceberg = async () => {
+  try {
+    const List = await fetchIcebergV2({
+      route: '/okms/resource',
+    }).then(({ data, status }) => ({ data, status }));
+    return List.data as OKMS[];
+  } catch (error) {
+    return null;
+  }
+};
+
+export const paginateResults = (items: OKMS[], pagination: PaginationState) => {
+  return {
+    rows: items.slice(
+      pagination.pageIndex * pagination.pageSize,
+      (pagination.pageIndex + 1) * pagination.pageSize,
+    ),
+    pageCount: Math.ceil(items.length / pagination.pageSize),
+    totalRows: items.length,
+  };
+};
```

Comment:
You might don't need it as apiv2 still doesn't manage `pageIndex` and `totalCount`